src/libraries/System.Text.Encodings.Web/src/PACKAGE.md

```diff
@@ -2,43 +2,131 @@
 
 <!-- A description of the package and where one can find more documentation -->
 
+Provides types for encoding and escaping strings for use in JavaScript, HTML, and URLs.
 
+This package is essential for protecting web applications against cross-site scripting (XSS) attacks by safely encoding text, and it offers extensive support for Unicode, allowing fine-grained control over which characters are encoded and which are left unescaped.
 
 ## Key Features
 
 <!-- The key features of this package -->
 
-*
-*
-*
+* Safe encoders for HTML, JavaScript, and URL strings.
+* Extensible to support custom encoding scenarios, including the ability to specify Unicode ranges.
+* Helps prevent cross-site scripting (XSS) vulnerabilities.
+* Flexible Unicode encoding with support for specifying individual or predefined ranges to cover broader sets of characters, including options to avoid escaping specific language character sets.
 
 ## How to Use
 
 <!-- A compelling example on how to use this package with code, as well as any specific guidelines for when to use the package -->
 
+### Encoding HTML, JavaScript, and URLs
+
+```csharp
+using System.Text.Encodings.Web;
+
+string unsafeString = "<script>alert('XSS Attack!');</script>";
+
+// HTML encode the string to safely display it on a web page.
+string safeHtml = HtmlEncoder.Default.Encode(unsafeString);
+Console.WriteLine(safeHtml);
+// &lt;script&gt;alert(&#x27;XSS Attack!&#x27;);&lt;/script&gt;
+
+// JavaScript encode the string to safely include it in a JavaScript context.
+string safeJavaScript = JavaScriptEncoder.Default.Encode(unsafeString);
+Console.WriteLine(safeJavaScript);
+// \u003Cscript\u003Ealert(\u0027XSS Attack!\u0027);\u003C/script\u003E
+
+string urlPart = "user input with spaces and & symbols";
+
+// URL encode the string to safely include it in a URL.
+string encodedUrlPart = UrlEncoder.Default.Encode(urlPart);
+Console.WriteLine(encodedUrlPart);
+// user%20input%20with%20spaces%20and%20%26%20symbols
+```
+
+### Custom Encoding Scenario with Specific Unicode Ranges
+
+```csharp
+using System.Text.Encodings.Web;
+using System.Text.Unicode;
+
+TextEncoderSettings customEncoderSettings = new TextEncoderSettings();
```

Comment:
nit: it would be good to use the simplified new operator on the right hand side: `new();`